### DISCUSSION TITLE
Plot peakMatrix in embedding

### DISCUSSION THREAD

**Discussion by Zepeng-Mu** [GreenleafLab/ArchR#1421] [Questions / Documentation] [answered]
Hi, I'm wondering if there is a way to plot a peak using `plotEmbedding`, like what we do with genes when using `GeneScoreMatrix`. Right now I cannot plot using `peakMatrix` because `getFeatures()` returns `NULL`. Thanks!!

**Comment by rcorces** [comment, marked as answer]
Yes - I've come across this before (https://github.com/GreenleafLab/ArchR/issues/308#issuecomment-792338095)

Im not sure what the original rationale was for not having a `name` in `PeakMatrix` and whether there will be unforeseen downstream consequences. I'm also not sure what the ideal "name" of a peak should be.

For what its worth, you can get around this. For example:

```
ps <- projHeme5@peakSet
ps$name <- paste0(seqnames(ps),"_peak",ps$idx)
projHeme5@peakSet <- ps
projHeme5 <- addPeakMatrix(ArchRProj = projHeme5, force = TRUE, threads = 16)
plotEmbedding(
  ArchRProj = projHeme5,
  colorBy = "PeakMatrix",
  name = c("chr1_peak300"),
  embedding = "UMAP",
  quantCut = c(0.01, 0.95),
  imputeWeights = getImputeWeights(ArchRProj = projHeme5),
  rastr = TRUE,
  plotAs = "points"
)
```
![image](https://user-images.githubusercontent.com/10343046/167764126-22edbb95-dd3e-4cc3-8190-efe597a89770.png)

For something more stable - I've added a line to `addPeakSet()` which should automatically add a `name` column to new peak sets.
https://github.com/GreenleafLab/ArchR/commit/2dfc240e26e12c89eef6e2161c7aa4e47e81f693 in `dev_peakName`.
This works for me on `dev_peakName`. Let me know if it solves your problem.
```
ps <- projHeme5@peakSet
ps$name <- NULL
projHeme5 <- addPeakSet(ArchRProj = projHeme5, peakSet = ps, force = TRUE)
projHeme5 <- addPeakMatrix(ArchRProj = projHeme5, force = TRUE)
plotEmbedding(
  ArchRProj = projHeme5,
  colorBy = "PeakMatrix",
  name = c("chr1_peak300"),
  embedding = "UMAP",
  quantCut = c(0.01, 0.95),
  imputeWeights = getImputeWeights(ArchRProj = projHeme5),
  rastr = TRUE,
  plotAs = "points"
)
```

**Reply by Zepeng-Mu** [reply]
Thanks a lot! The first solution should work for me. Since I also use the peaks in other tools, I've been defining my own peak names, so I might just keep using my own names for now.
